blazar/enforcement/filters/base_filter.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2013 Bull.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseFilter:
+
+    enforcement_opts = []
+
+    def __init__(self, conf=None):
+        self.conf = conf
+
+        for opt in self.enforcement_opts:
+            self.conf.register_opt(opt, 'enforcement')
```

Comment:
In the spec we outlined that some filters may define their own group. E.g., the external filter has its options all under a new `enforcement_external` group. Not sure what the best way to express that is, maybe the filter can define a `enforcement_opt_group`, which defaults to `'enforcement'`?